ThePensionsRegulator.Umbraco/TokenList.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System.Collections;
+using System.Text.RegularExpressions;
+using Umbraco.Cms.Core.Models.PublishedContent;
+using Umbraco.Extensions;
+
+namespace ThePensionsRegulator.Umbraco
+{
+	/// <summary>
+	/// Exposes a property of an <see cref="IOverridablePublishedElement"/> as a list of tokens.
+	/// </summary>
+	public class TokenList : IList<string>
+	{
+		private readonly IPublishedElement? _publishedElement;
+		private readonly IOverridablePublishedElement? _overridablePublishedElement;
+		private readonly string _propertyName;
+		private readonly List<string> _tokens = new();
+		private bool _tokensCurrent = false;
+		private Regex _regex = new Regex(@"\s+", RegexOptions.Compiled);
+		private readonly bool _isReadOnly;
+		private readonly string _separator;
+
+		/// <summary>
+		/// Creates a new <see cref="TokenList"/>
+		/// </summary>
+		/// <param name="publishedElement">The Umbraco content or element containing the property with the tokenised value.</param>
+		/// <param name="propertyName">The name of the Umbraco property where the tokens are stored.</param>
+		/// <param name="separator">The string with which tokens are separated in the property value.</param>
+		/// <exception cref="ArgumentException">Thrown is <c>propertyName</c> is <c>null</c> or an empty string.</exception>
+		public TokenList(IPublishedElement? publishedElement, string propertyName, string separator = " ")
+		{
+			if (string.IsNullOrEmpty(propertyName))
+			{
+				throw new ArgumentException($"'{nameof(propertyName)}' cannot be null or empty.", nameof(propertyName));
+			}
+
+			_publishedElement = publishedElement;
+			_overridablePublishedElement = _publishedElement as IOverridablePublishedElement;
+			_isReadOnly = _overridablePublishedElement == null;
+			_propertyName = propertyName;
+			_separator = separator;
+		}
+
+		private void Tokenise()
+		{
+			var raw = _isReadOnly ? _publishedElement?.Value<string>(_propertyName) : _overridablePublishedElement!.Value<string>(_propertyName);
+			_tokens.Clear();
+			if (!string.IsNullOrEmpty(raw))
+			{
+				raw = _regex.Replace(raw, _separator).Trim();
+				_tokens.AddRange(raw.Split(_separator));
+			}
+			else
```

Comment:
Is this `else` really needed? The list will already be empty if this `else` statement is reached

Reply:
Well spotted. Fixed.